plugins/hypervisors/hyperv/src/com/cloud/hypervisor/hyperv/resource/HypervDirectConnectResource.java

```diff
@@ -595,6 +597,25 @@ private PlugNicAnswer execute(PlugNicCommand cmd) {
         }
     }
 
+    private CopyFileInVmAnswer execute(CopyFileInVmCommand cmd) {
+        File keyFile = getSystemVMKeyFile();
+        try {
+            File file = new File(cmd.getSrc());
+            if(file.exists()) {
+                if(file.isDirectory()) {
+                    for (File f : FileUtils.listFiles(new File(cmd.getSrc()), TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE)) {
+                        SshHelper.scpTo(cmd.getVmIp(), 3922, "root", keyFile, null, cmd.getDest(), f.getCanonicalPath(), null);
+                    }
+                } else {
+                    SshHelper.scpTo(cmd.getVmIp(), 3922, "root", keyFile, null, cmd.getDest(), file.getCanonicalPath(), null);
+                }
+            }
+        } catch (Exception e) {
+            s_logger.error("Fail to copy file " + cmd.getSrc() + " in VM " + cmd.getVmIp(), e);
+            return new CopyFileInVmAnswer(cmd, e);
+        }
+        return new CopyFileInVmAnswer(cmd);
```

Comment:
I mean in case a copyfilein..if moved inside try block throws an exception , then the code return new...copyfile inside catch still works.